### ISSUE TITLE
Running the elasticsearch-operator on minikube

### ISSUE THREAD

**Issue by secat** [upmc-enterprises/elasticsearch-operator#7]
Is it possible to use the `elasticsearch-operator` on `minikube`?

I used the [example/controller.yaml](https://github.com/upmc-enterprises/elasticsearch-operator/blob/master/example/controller.yaml) manifests with the following one:
```
apiVersion: enterprises.upmc.com/v1
kind: ElasticsearchCluster
metadata:
  name: elasticsearch-logstack
spec:
  client-node-replicas: 1
  master-node-replicas: 2
  data-node-replicas: 1
  data-volume-size: 5Gi
```

Here is the error I have on the `es-client` pod (same with the `es-master` pod):
```
MountVolume.SetUp failed for volume "kubernetes.io/secret/a826364d-eec3-11e6-8689-0800275f7578-es-certs" (spec.Name: "es-certs") pod "a826364d-eec3-11e6-8689-0800275f7578" (UID: "a826364d-eec3-11e6-8689-0800275f7578") with: secrets "es-certs" not found
```

Here are the operator logs:
```
2017-02-09T12:18:05.160068721Z time="2017-02-09T12:18:05Z" level=info msg="elasticsearch operator starting up!" 
2017-02-09T12:18:05.160120760Z time="2017-02-09T12:18:05Z" level=info msg="Using Variables:" 
2017-02-09T12:18:05.160127036Z time="2017-02-09T12:18:05Z" level=info msg="   baseImage: upmcenterprises/docker-elasticsearch-kubernetes:5.1.1" 
2017-02-09T12:18:05.160134558Z time="2017-02-09T12:18:05Z" level=info msg="Using InCluster k8s config" 
2017-02-09T12:18:05.191058689Z time="2017-02-09T12:18:05Z" level=info msg="ElasticSearchCluster ThirdPartyResource not found, creating..." 
2017-02-09T12:18:05.193884371Z time="2017-02-09T12:18:05Z" level=error msg="Get http://127.0.0.1:8001/apis/enterprises.upmc.com/v1/namespaces/oversea-experimental/elasticsearchclusters: dial tcp 127.0.0.1:8001: getsockopt: connection refused" 
2017-02-09T12:18:10.199308164Z time="2017-02-09T12:18:10Z" level=info msg="Found 0 existing clusters " 
2017-02-09T12:18:10.199455077Z time="2017-02-09T12:18:10Z" level=info msg="Watching for elastic search events..." 
2017-02-09T12:31:18.610324812Z time="2017-02-09T12:31:18Z" level=info msg="--------> ElasticSearch Event!" 
2017-02-09T12:31:18.610369906Z time="2017-02-09T12:31:18Z" level=info msg="Using [upmcenterprises/docker-elasticsearch-kubernetes:5.1.1] as image for es cluster" 
2017-02-09T12:31:18.612809266Z time="2017-02-09T12:31:18Z" level=info msg="Discovery Service not found, creating..." 
2017-02-09T12:31:18.663150063Z time="2017-02-09T12:31:18Z" level=info msg="es-data-svc not found, creating..." 
2017-02-09T12:31:18.694848768Z time="2017-02-09T12:31:18Z" level=info msg="elasticsearch not found, creating..." 
2017-02-09T12:31:18.735464044Z time="2017-02-09T12:31:18Z" level=info msg="es-client not found, creating..." 
2017-02-09T12:31:18.765675568Z time="2017-02-09T12:31:18Z" level=info msg="es-master not found, creating..." 
2017-02-09T12:31:18.840863773Z time="2017-02-09T12:31:18Z" level=info msg="StorgeClass es-default not found, creating..." 
2017-02-09T12:31:18.862609386Z time="2017-02-09T12:31:18Z" level=error msg="Could not create storage class: StorageClass.storage.k8s.io \"es-default\" is invalid: provisioner: Required value" 
2017-02-09T12:31:18.901527823Z time="2017-02-09T12:31:18Z" level=info msg="StatefulSet es-data-es-default not found, creating..." 
```

Here is the `PersistentVolumeClaims` created by the operator:
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  annotations:
    volume.beta.kubernetes.io/storage-class: es-default
  creationTimestamp: 2017-02-09T12:31:19Z
  labels:
    component: elasticsearch
    name: es-data-es-default
    role: data
  name: es-data-es-data-es-default-0
  namespace: oversea-experimental
  resourceVersion: "162655"
  selfLink: /api/v1/namespaces/oversea-experimental/persistentvolumeclaims/es-data-es-data-es-default-0
  uid: a8514033-eec3-11e6-8689-0800275f7578
spec:
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
status:
  phase: Pending
```

However the `volume.beta.kubernetes.io/storage-class` annotation currently does not work on `minikube`, but the `volume.alpha.kubernetes.io/storage-class` works usually.

**Comment by stevesloka** [comment]
Hey @secat, I think I just about have this working. I've got to test it out a bit more, but should have this soon. Thanks!

**Comment by stevesloka** [comment]
Resolved in #16

**Comment by vainikkaj** [comment]
Hi @stevesloka, I gave this a spin in minikube but ended with same error message (`secrets "es-certs" not found`)
